### ISSUE TITLE
Fix S1764 FP: Don't run on Test code

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#4659]
This rule should not run on test code. Equals is often subject of the test itself. 

S1764 Change one instance of 'context1' to a different value; comparing 'context1' to itself always 
```
context1.Equals(context1).Should().BeTrue(); 
```

As of now, the issue is raised in SL/NuGet context.

**Comment by pavel-mikula-sonarsource** [comment]
The PR to review is the one in the `rspec` (linked above)
